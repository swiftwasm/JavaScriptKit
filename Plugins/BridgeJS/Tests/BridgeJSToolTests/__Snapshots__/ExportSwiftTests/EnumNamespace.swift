// NOTICE: This is auto-generated code by BridgeJS from JavaScriptKit,
// DO NOT EDIT.
//
// To update this file, just rebuild your project or run
// `swift package bridge-js`.

@_spi(BridgeJS) import JavaScriptKit

extension Networking.API.Method {
    @_spi(BridgeJS) @_transparent public consuming func bridgeJSLowerParameter() -> Int32 {
        return bridgeJSRawValue
    }
    @_spi(BridgeJS) @_transparent public static func bridgeJSLiftReturn(_ value: Int32) -> Networking.API.Method {
        return Networking.API.Method(bridgeJSRawValue: value)!
    }
    @_spi(BridgeJS) @_transparent public static func bridgeJSLiftParameter(_ value: Int32) -> Networking.API.Method {
        return Networking.API.Method(bridgeJSRawValue: value)!
    }
    @_spi(BridgeJS) @_transparent public consuming func bridgeJSLowerReturn() -> Int32 {
        return bridgeJSRawValue
    }

    private init?(bridgeJSRawValue: Int32) {
        switch bridgeJSRawValue {
        case 0:
            self = .get
        case 1:
            self = .post
        case 2:
            self = .put
        case 3:
            self = .delete
        default:
            return nil
        }
    }

    private var bridgeJSRawValue: Int32 {
        switch self {
        case .get:
            return 0
        case .post:
            return 1
        case .put:
            return 2
        case .delete:
            return 3
        }
    }
}

extension Configuration.LogLevel: _BridgedSwiftEnumNoPayload {
}

extension Configuration.Port: _BridgedSwiftEnumNoPayload {
}

extension Internal.SupportedMethod {
    @_spi(BridgeJS) @_transparent public consuming func bridgeJSLowerParameter() -> Int32 {
        return bridgeJSRawValue
    }
    @_spi(BridgeJS) @_transparent public static func bridgeJSLiftReturn(_ value: Int32) -> Internal.SupportedMethod {
        return Internal.SupportedMethod(bridgeJSRawValue: value)!
    }
    @_spi(BridgeJS) @_transparent public static func bridgeJSLiftParameter(_ value: Int32) -> Internal.SupportedMethod {
        return Internal.SupportedMethod(bridgeJSRawValue: value)!
    }
    @_spi(BridgeJS) @_transparent public consuming func bridgeJSLowerReturn() -> Int32 {
        return bridgeJSRawValue
    }

    private init?(bridgeJSRawValue: Int32) {
        switch bridgeJSRawValue {
        case 0:
            self = .get
        case 1:
            self = .post
        default:
            return nil
        }
    }

    private var bridgeJSRawValue: Int32 {
        switch self {
        case .get:
            return 0
        case .post:
            return 1
        }
    }
}

@_expose(wasm, "bjs_Converter_init")
@_cdecl("bjs_Converter_init")
public func _bjs_Converter_init() -> UnsafeMutableRawPointer {
    #if arch(wasm32)
    let ret = Utils.Converter()
    return ret.bridgeJSLowerReturn()
    #else
    fatalError("Only available on WebAssembly")
    #endif
}

@_expose(wasm, "bjs_Converter_toString")
@_cdecl("bjs_Converter_toString")
public func _bjs_Converter_toString(_self: UnsafeMutableRawPointer, value: Int32) -> Void {
    #if arch(wasm32)
    let ret = Utils.Converter.bridgeJSLiftParameter(_self).toString(value: Int.bridgeJSLiftParameter(value))
    return ret.bridgeJSLowerReturn()
    #else
    fatalError("Only available on WebAssembly")
    #endif
}

@_expose(wasm, "bjs_Converter_deinit")
@_cdecl("bjs_Converter_deinit")
public func _bjs_Converter_deinit(pointer: UnsafeMutableRawPointer) {
    Unmanaged<Utils.Converter>.fromOpaque(pointer).release()
}

extension Utils.Converter: ConvertibleToJSValue, _BridgedSwiftHeapObject {
    var jsValue: JSValue {
        #if arch(wasm32)
        @_extern(wasm, module: "TestModule", name: "bjs_Converter_wrap")
        func _bjs_Converter_wrap(_: UnsafeMutableRawPointer) -> Int32
        #else
        func _bjs_Converter_wrap(_: UnsafeMutableRawPointer) -> Int32 {
            fatalError("Only available on WebAssembly")
        }
        #endif
        return .object(JSObject(id: UInt32(bitPattern: _bjs_Converter_wrap(Unmanaged.passRetained(self).toOpaque()))))
    }
}

@_expose(wasm, "bjs_HTTPServer_init")
@_cdecl("bjs_HTTPServer_init")
public func _bjs_HTTPServer_init() -> UnsafeMutableRawPointer {
    #if arch(wasm32)
    let ret = Networking.API.HTTPServer()
    return ret.bridgeJSLowerReturn()
    #else
    fatalError("Only available on WebAssembly")
    #endif
}

@_expose(wasm, "bjs_HTTPServer_call")
@_cdecl("bjs_HTTPServer_call")
public func _bjs_HTTPServer_call(_self: UnsafeMutableRawPointer, method: Int32) -> Void {
    #if arch(wasm32)
    Networking.API.HTTPServer.bridgeJSLiftParameter(_self).call(_: Networking.API.Method.bridgeJSLiftParameter(method))
    #else
    fatalError("Only available on WebAssembly")
    #endif
}

@_expose(wasm, "bjs_HTTPServer_deinit")
@_cdecl("bjs_HTTPServer_deinit")
public func _bjs_HTTPServer_deinit(pointer: UnsafeMutableRawPointer) {
    Unmanaged<Networking.API.HTTPServer>.fromOpaque(pointer).release()
}

extension Networking.API.HTTPServer: ConvertibleToJSValue, _BridgedSwiftHeapObject {
    var jsValue: JSValue {
        #if arch(wasm32)
        @_extern(wasm, module: "TestModule", name: "bjs_HTTPServer_wrap")
        func _bjs_HTTPServer_wrap(_: UnsafeMutableRawPointer) -> Int32
        #else
        func _bjs_HTTPServer_wrap(_: UnsafeMutableRawPointer) -> Int32 {
            fatalError("Only available on WebAssembly")
        }
        #endif
        return .object(JSObject(id: UInt32(bitPattern: _bjs_HTTPServer_wrap(Unmanaged.passRetained(self).toOpaque()))))
    }
}

@_expose(wasm, "bjs_TestServer_init")
@_cdecl("bjs_TestServer_init")
public func _bjs_TestServer_init() -> UnsafeMutableRawPointer {
    #if arch(wasm32)
    let ret = Internal.TestServer()
    return ret.bridgeJSLowerReturn()
    #else
    fatalError("Only available on WebAssembly")
    #endif
}

@_expose(wasm, "bjs_TestServer_call")
@_cdecl("bjs_TestServer_call")
public func _bjs_TestServer_call(_self: UnsafeMutableRawPointer, method: Int32) -> Void {
    #if arch(wasm32)
    Internal.TestServer.bridgeJSLiftParameter(_self).call(_: Internal.SupportedMethod.bridgeJSLiftParameter(method))
    #else
    fatalError("Only available on WebAssembly")
    #endif
}

@_expose(wasm, "bjs_TestServer_deinit")
@_cdecl("bjs_TestServer_deinit")
public func _bjs_TestServer_deinit(pointer: UnsafeMutableRawPointer) {
    Unmanaged<Internal.TestServer>.fromOpaque(pointer).release()
}

extension Internal.TestServer: ConvertibleToJSValue, _BridgedSwiftHeapObject {
    var jsValue: JSValue {
        #if arch(wasm32)
        @_extern(wasm, module: "TestModule", name: "bjs_TestServer_wrap")
        func _bjs_TestServer_wrap(_: UnsafeMutableRawPointer) -> Int32
        #else
        func _bjs_TestServer_wrap(_: UnsafeMutableRawPointer) -> Int32 {
            fatalError("Only available on WebAssembly")
        }
        #endif
        return .object(JSObject(id: UInt32(bitPattern: _bjs_TestServer_wrap(Unmanaged.passRetained(self).toOpaque()))))
    }
}