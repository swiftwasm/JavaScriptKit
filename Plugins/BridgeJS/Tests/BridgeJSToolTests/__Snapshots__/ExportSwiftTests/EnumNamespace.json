{
  "classes" : [
    {
      "constructor" : {
        "abiName" : "bjs_Converter_init",
        "effects" : {
          "isAsync" : false,
          "isStatic" : false,
          "isThrows" : false
        },
        "parameters" : [

        ]
      },
      "methods" : [
        {
          "abiName" : "bjs_Converter_toString",
          "effects" : {
            "isAsync" : false,
            "isStatic" : false,
            "isThrows" : false
          },
          "name" : "toString",
          "parameters" : [
            {
              "label" : "value",
              "name" : "value",
              "type" : {
                "int" : {

                }
              }
            }
          ],
          "returnType" : {
            "string" : {

            }
          }
        }
      ],
      "name" : "Converter",
      "namespace" : [
        "Utils"
      ],
      "properties" : [

      ],
      "swiftCallName" : "Utils.Converter"
    },
    {
      "constructor" : {
        "abiName" : "bjs_HTTPServer_init",
        "effects" : {
          "isAsync" : false,
          "isStatic" : false,
          "isThrows" : false
        },
        "parameters" : [

        ]
      },
      "methods" : [
        {
          "abiName" : "bjs_HTTPServer_call",
          "effects" : {
            "isAsync" : false,
            "isStatic" : false,
            "isThrows" : false
          },
          "name" : "call",
          "parameters" : [
            {
              "label" : "_",
              "name" : "method",
              "type" : {
                "caseEnum" : {
                  "_0" : "Networking.API.Method"
                }
              }
            }
          ],
          "returnType" : {
            "void" : {

            }
          }
        }
      ],
      "name" : "HTTPServer",
      "namespace" : [
        "Networking",
        "API"
      ],
      "properties" : [

      ],
      "swiftCallName" : "Networking.API.HTTPServer"
    },
    {
      "constructor" : {
        "abiName" : "bjs_TestServer_init",
        "effects" : {
          "isAsync" : false,
          "isStatic" : false,
          "isThrows" : false
        },
        "parameters" : [

        ]
      },
      "methods" : [
        {
          "abiName" : "bjs_TestServer_call",
          "effects" : {
            "isAsync" : false,
            "isStatic" : false,
            "isThrows" : false
          },
          "name" : "call",
          "parameters" : [
            {
              "label" : "_",
              "name" : "method",
              "type" : {
                "caseEnum" : {
                  "_0" : "Internal.SupportedMethod"
                }
              }
            }
          ],
          "returnType" : {
            "void" : {

            }
          }
        }
      ],
      "name" : "TestServer",
      "namespace" : [
        "Networking",
        "APIV2",
        "Internal"
      ],
      "properties" : [

      ],
      "swiftCallName" : "Internal.TestServer"
    }
  ],
  "enums" : [
    {
      "cases" : [

      ],
      "emitStyle" : "const",
      "name" : "Utils",
      "staticMethods" : [

      ],
      "swiftCallName" : "Utils"
    },
    {
      "cases" : [
        {
          "associatedValues" : [

          ],
          "name" : "get"
        },
        {
          "associatedValues" : [

          ],
          "name" : "post"
        },
        {
          "associatedValues" : [

          ],
          "name" : "put"
        },
        {
          "associatedValues" : [

          ],
          "name" : "delete"
        }
      ],
      "emitStyle" : "const",
      "name" : "Method",
      "namespace" : [
        "Networking",
        "API"
      ],
      "staticMethods" : [

      ],
      "swiftCallName" : "Networking.API.Method"
    },
    {
      "cases" : [
        {
          "associatedValues" : [

          ],
          "name" : "debug",
          "rawValue" : "debug"
        },
        {
          "associatedValues" : [

          ],
          "name" : "info",
          "rawValue" : "info"
        },
        {
          "associatedValues" : [

          ],
          "name" : "warning",
          "rawValue" : "warning"
        },
        {
          "associatedValues" : [

          ],
          "name" : "error",
          "rawValue" : "error"
        }
      ],
      "emitStyle" : "const",
      "name" : "LogLevel",
      "namespace" : [
        "Configuration"
      ],
      "rawType" : "String",
      "staticMethods" : [

      ],
      "swiftCallName" : "Configuration.LogLevel"
    },
    {
      "cases" : [
        {
          "associatedValues" : [

          ],
          "name" : "http",
          "rawValue" : "80"
        },
        {
          "associatedValues" : [

          ],
          "name" : "https",
          "rawValue" : "443"
        },
        {
          "associatedValues" : [

          ],
          "name" : "development",
          "rawValue" : "3000"
        }
      ],
      "emitStyle" : "const",
      "name" : "Port",
      "namespace" : [
        "Configuration"
      ],
      "rawType" : "Int",
      "staticMethods" : [

      ],
      "swiftCallName" : "Configuration.Port"
    },
    {
      "cases" : [
        {
          "associatedValues" : [

          ],
          "name" : "get"
        },
        {
          "associatedValues" : [

          ],
          "name" : "post"
        }
      ],
      "emitStyle" : "const",
      "name" : "SupportedMethod",
      "namespace" : [
        "Networking",
        "APIV2",
        "Internal"
      ],
      "staticMethods" : [

      ],
      "swiftCallName" : "Internal.SupportedMethod"
    }
  ],
  "functions" : [

  ],
  "moduleName" : "TestModule"
}